app/controllers/testOnly/TestOnlyRouterController.scala

```diff
@@ -31,8 +31,17 @@ class TestOnlyRouterController @Inject()(
 )(implicit val ec: ExecutionContext)
     extends BackendController(cc) {
 
-  def handleMessage(): Action[NodeSeq] = Action.async(parse.xml) {
+  def fromCoreMessage: Action[NodeSeq] = Action.async(parse.xml) {
     implicit request =>
-      connector.sendMessage(request.body, request.headers).map(response => Status(response.status))
+      connector
```

Comment:
Is the static reference to Action here throwing up any warnings? If it is then we can inject the `DefaultActionBuilder` and use that to build our actions